### PR TITLE
Support logging using log4j2 with slf4j

### PR DESCRIPTION
**Pull request by jedrazb**
- unify logging, replace all uses of `java.util.logging` with `slf4j` interface
- use `log4j2` as logging backend
- provide simple log4j2 config
- also adding wildcard expression for classpath: it will use all jars dropped to `lib` folder when running application. Previously classpath was including only dependencies defined in gradle.

Example generated logs when starting the service:
```
[INFO ] 2021-05-21 09:21:57.173 [main] ThreadPoolExecutorFactory - Creating LuceneIndexingExecutor of size 17
[INFO ] 2021-05-21 09:21:57.176 [main] ThreadPoolExecutorFactory - Creating LuceneSearchExecutor of size 25
[INFO ] 2021-05-21 09:21:57.197 [main] Plugin - Loading plugins: []
[INFO ] 2021-05-21 09:21:57.309 [main] ThreadPoolExecutorFactory - Creating GrpcLuceneServerExecutor of size 17
[INFO ] 2021-05-21 09:21:57.407 [main] LuceneServer - Server started, listening on 6000 for messages
[INFO ] 2021-05-21 09:21:57.417 [main] ThreadPoolExecutorFactory - Creating GrpcReplicationServerExecutor of size 17
[INFO ] 2021-05-21 09:21:57.419 [main] LuceneServer - Server started, listening on 6001 for replication messages

```

### CODE REVIEW

File: src/main/java/com/yelp/nrtsearch/server/utils/ParallelDocumentIndexer.java [hunk 1]
```diff
@@ -86,10 +87,8 @@
                     oneDocBuilder, rawLines.stream(), luceneServerClient));
         return genIdFuture;
       } catch (RejectedExecutionException e) {
-        logger.log(
-            Level.WARNING,
-            String.format("Waiting for 1s for LinkedBlockingQueue to have more capacity"),
-            e);
+        logger.warn(
+            String.format("Waiting for 1s for LinkedBlockingQueue to have more capacity"), e);
```
Comment (sarthakn7) [inline src/main/java/com/yelp/nrtsearch/server/utils/ParallelDocumentIndexer.java:91]: Can we replace the `String.format` calls with the way slf4j does formatting? I think intellij might offer the option to replace the usage

File: build.gradle [hunk 2]
```diff
@@ -47,9 +49,12 @@ dependencies {
     implementation "io.prometheus:simpleclient_servlet:${prometheusClientVersion}"
     implementation "io.prometheus:simpleclient_hotspot:${prometheusClientVersion}"
 
-    //
+    //logging deps
     implementation "org.slf4j:slf4j-api:${project.ext.slf4jVersion}"
-    implementation "org.slf4j:slf4j-jdk14:2.0.0-alpha1"
+    implementation "org.apache.logging.log4j:log4j-slf4j18-impl:${log4jVersion}"
+    implementation "org.apache.logging.log4j:log4j-api:${log4jVersion}"
+    implementation "org.apache.logging.log4j:log4j-core:${log4jVersion}"
```
Comment (sarthakn7) [inline build.gradle:56]: You also need to add the dependency for lmax disruptor to make the logging async http://logging.apache.org/log4j/2.x/manual/async.html

File: build.gradle [hunk 3]
```diff
@@ -59,6 +64,9 @@ dependencies {
     implementation "javax.xml.bind:jaxb-api:2.3.1"
     implementation "com.guicedee.services:guice:${guicedeeVersion}"
     implementation "org.lz4:lz4-java:${lz4Version}"
+    implementation "com.fasterxml.jackson.core:jackson-databind:${jacksonYamlVersion}"
+    implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:${jacksonYamlVersion}"
```
Comment (sarthakn7) [inline build.gradle:68]: Are the jackson dependencies required? I think log4j2 can read the configs without it

File: src/test/java/com/yelp/nrtsearch/server/YelpReviewsTest.java [hunk 4]
```diff
@@ -357,10 +357,10 @@ public static void main(String[] args) throws IOException, InterruptedException
       logger.info("done...");
 
     } catch (StatusRuntimeException e) {
-      logger.severe("RPC failed with status " + e.getStatus());
+      logger.error("RPC failed with status " + e.getStatus());
```
Comment (sarthakn7) [inline src/test/java/com/yelp/nrtsearch/server/YelpReviewsTest.java:360]: For these too if possible let's use slf4j formatting

File: build.gradle (hunk 3 above)
Comment (jedrazb) [reply build.gradle:68]: Yup, they are required: https://logging.apache.org/log4j/2.x/runtime-dependencies.html (search for YAML) Without this package log4j2 is unable to read yaml config file

**Review by sarthakn7 — APPROVED** [review]

**Review by sarthakn7 — APPROVED** [review]

**Pull request merged**